File: prid1920-g01/Models/Comment.cs
using System;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;

namespace prid1920_g01.Models
{

    public class Comment
    {

        [Key]
        public int Id { get; set; }
        [Required(ErrorMessage = "Required Body")]
        public string Body { get; set; }
        [Required(ErrorMessage = "Required Timestamp")]
        public DateTime Timestamp { get; set; } = DateTime.Now;
        [Required(ErrorMessage = "Required User")]
        public int UserId { get; set; }
        public virtual User User { get; set; }
        [Required(ErrorMessage = "Required Post")]
        public int PostId { get; set; }
        public virtual Post Post { get; set; }

    }
}

File: prid1920-g01/ClientApp/src/app/models/comment.ts
import { User } from "./user";
import { Post } from "./post";

export class Comment {
    id: string;
    body: string;
    timestamp: string;
    userId: string;
    user: User;
    postId: string;
    post: Post;

    constructor(data: any) {
        this.id = data.id;
        this.body = data.body;
        this.timestamp = data.timestamp;
        this.userId = data.userId;
        this.user = data.user;
        this.postId = data.postId;
        this.post = data.post;
    }
}